server/controllers/reaction.js

```diff
@@ -0,0 +1,72 @@
+import reactionRepo from '../repository/reactionRepository';
+import articleRepo from '../repository/articleRepository';
+import { goodHttpResponse, badHttpResponse, } from '../utilities/httpResponse';
+
+/**
+ * Reaction Controller class
+ */
+class Reaction {
+  /**
+   * Like Article
+   * @param {object} request Request Object
+   * @param {object} response Response Object
+   * @returns {object} User Object
+   */
+  static async postReaction(request, response) {
+    const { slug } = request.params;
+    const { userId } = request;
+    const { reactionType } = request.body;
+
+    let message;
+    switch (reactionType) {
+      case 'Like':
+        message = 'You liked this article';
+        break;
+      case 'Dislike':
+        message = 'You disliked this article';
+        break;
+      case 'Love':
+        message = 'You loved this article';
+        break;
+      default:
+        message = '';
+    }
+    try {
+      const existingArticle = await articleRepo.getArticleBySlug(slug);
+      if (!existingArticle) {
+        return badHttpResponse(response, 404, 'article does not exist');
+      }
+      const existingReaction = await reactionRepo.getReaction(userId, existingArticle.id);
+
+      if (existingReaction) {
```

Comment:
This can be simplified to be `existingReaction && existingReaction.reactionType !== reactionType`

Reply:
@seyi-adeleke , I'm making two checks there, if the reaction type sent is different from existing reaction, it should update it, but if the reaction type sent is the same as the existing reaction, it should be removed. its a nested IFs.